pkg/collector/corechecks/systemd/systemd.go

```diff
@@ -0,0 +1,451 @@
+	initConf systemdInitConfig
+}
+
+type systemdStats interface {
+	// Dbus Connection
+	NewConn() (*dbus.Conn, error)
+	CloseConn(c *dbus.Conn)
+
+	// System Data
+	SystemState(c *dbus.Conn) (*dbus.Property, error)
+	ListUnits(c *dbus.Conn) ([]dbus.UnitStatus, error)
+	GetUnitTypeProperties(c *dbus.Conn, unitName string, unitType string) (map[string]interface{}, error)
+
+	// Misc
+	TimeNanoNow() int64
+}
+
+type defaultSystemdStats struct{}
+
+func (s *defaultSystemdStats) NewConn() (*dbus.Conn, error) {
+	return dbus.New()
+}
+
+func (s *defaultSystemdStats) CloseConn(c *dbus.Conn) {
+	c.Close()
+}
+
+func (s *defaultSystemdStats) SystemState(c *dbus.Conn) (*dbus.Property, error) {
+	return c.SystemState()
+}
+
+func (s *defaultSystemdStats) ListUnits(conn *dbus.Conn) ([]dbus.UnitStatus, error) {
+	return conn.ListUnits()
+}
+
+func (s *defaultSystemdStats) GetUnitTypeProperties(c *dbus.Conn, unitName string, unitType string) (map[string]interface{}, error) {
+	return c.GetUnitTypeProperties(unitName, unitType)
+}
+
+func (s *defaultSystemdStats) TimeNanoNow() int64 {
+	return time.Now().UnixNano()
+}
+
+// Run executes the check
+func (c *Check) Run() error {
+	sender, err := aggregator.GetSender(c.ID())
+	if err != nil {
+		return err
+	}
+
+	conn, err := c.getDbusConn(sender)
+	if err != nil {
+		return err
+	}
+	defer c.stats.CloseConn(conn)
+
+	err = c.submitMetrics(sender, conn)
+	if err != nil {
+		return err
+	}
+	sender.Commit()
+	return nil
+}
+
+func (c *Check) getDbusConn(sender aggregator.Sender) (*dbus.Conn, error) {
+	conn, err := c.stats.NewConn()
+	if err != nil {
+		newErr := fmt.Errorf("Cannot create a connection: %v", err)
+		sender.ServiceCheck(canConnectServiceCheck, metrics.ServiceCheckCritical, "", nil, newErr.Error())
+		return nil, newErr
+	}
+
+	systemStateProp, err := c.stats.SystemState(conn)
+	if err != nil {
+		newErr := fmt.Errorf("Err calling SystemState: %v", err)
+		sender.ServiceCheck(canConnectServiceCheck, metrics.ServiceCheckCritical, "", nil, newErr.Error())
+		return nil, newErr
+	}
+	sender.ServiceCheck(canConnectServiceCheck, metrics.ServiceCheckOK, "", nil, "")
+
+	serviceCheckStatus := metrics.ServiceCheckUnknown
+	systemState, ok := systemStateProp.Value.Value().(string)
+	if ok {
+		status, ok := systemdStatusMapping[systemState]
+		if ok {
+			serviceCheckStatus = status
+		}
+	}
+	sender.ServiceCheck(systemStateServiceCheck, serviceCheckStatus, "", nil, fmt.Sprintf("Systemd status is %v", systemStateProp.Value))
+	return conn, nil
+}
+
+func (c *Check) submitMetrics(sender aggregator.Sender, conn *dbus.Conn) error {
+	units, err := c.stats.ListUnits(conn)
+	if err != nil {
+		return fmt.Errorf("Error getting list of units: %v", err)
+	}
+
+	c.submitCountMetrics(sender, units)
+
+	loadedCount := 0
+	for _, unit := range units {
+		if unit.LoadState == unitLoadedState {
+			loadedCount++
+		}
+		if !c.isMonitored(unit.Name) {
+			continue
+		}
+		tags := []string{unitTag + ":" + unit.Name}
```

Comment:
Since this is used several times across the integration code, having a constant helps avoid typos and makes it more maintainable if we need to change the value (not very likely for tags).

But yes, hard coding it makes it easier to read. 

So ok, will make the change.